src/storage/mod.rs

```diff
@@ -365,6 +400,40 @@ where
         }
     }
 
+    /// Insert the provided value at `id` and adds `id` to the mask.
+    ///
+    /// # Safety
+    ///
+    /// May only be called if `id` is not present in the mask.
+    #[inline(always)]
+    unsafe fn not_present_insert(&mut self, id: Index, value: T) {
+        // SAFETY: The mask was previously empty, so it is safe to
+        // insert. We immediately add the value to the mask below and
+        // unwinding from the `insert` call means that we don't need to
+        // include the value in the mask. `BitSet::add` won't unwind on 32-bit
+        // and 64-bit platforms since OOM aborts and any overflow in capacity
```

Comment:
> `BitSet::add` won't unwind on 32-bit and 64-bit platforms since OOM aborts

Seems like this might not always be the case, e.g. see https://blog.rust-lang.org/2023/03/09/Rust-1.68.0.html

Reply:
Fixed